File: week12/SacramentPlanner/SacramentPlanner/Models/PlannerModel.cs
using System;
using System.Collections.Generic;
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace SacramentPlanner.Models
{
    public class PlannerModel
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string Id { get; set; }

        public string WardName { get; set; }
        public MemberModel Presiding { get; set; }
        public MemberModel Conducting { get; set; }
        public HymnModel OpeningHymn { get; set; }
        public MemberModel Invocation { get; set; }
        public bool WardBusiness { get; set; }
        public HymnModel SacramentHymn { get; set; }
        public bool SacramentPassing { get; set; }
        public List<TalkModel> Talks { get; set; }
        public MusicalNumberModel MusicalNumber { get; set; }
        public HymnModel ClosingHymn { get; set; }
        public MemberModel Benediction { get; set; }
        public HymnModel DismissalSong { get; set; }
        public DateTime SacramentDate { get; set; }

        public PlannerModel(
            string wardName,
            MemberModel presiding,
            MemberModel conducting,
            HymnModel openingHymn,
            MemberModel invocation,
            bool wardBusiness,
            HymnModel sacramentHymn,
            bool sacramentPassing,
            List<TalkModel> talks,
            MusicalNumberModel musicalNumber,
            HymnModel closingHymn,
            MemberModel benediction,
            HymnModel dismissalSong,
            DateTime sacramentDate)
        {
            WardName = wardName;
            Presiding = presiding;
            Conducting = conducting;
            OpeningHymn = openingHymn;
            Invocation = invocation;
            WardBusiness = wardBusiness;
            SacramentHymn = sacramentHymn;
            SacramentPassing = SacramentPassing;
            Talks = talks;
            MusicalNumber = musicalNumber;
            ClosingHymn = closingHymn;
            Benediction = benediction;
            DismissalSong = dismissalSong;
            SacramentDate = sacramentDate;
        }

        public PlannerModel() { }
    }
}

File: week12/SacramentPlanner/SacramentPlanner/ClientApp/src/app/planners/planner.model.ts
import { Member } from "../members/member.model";
import { Hymn } from "../hymns/hymn.model";
import { Talk } from "../talk.model";
import { MusicalNumber } from "../musicalNumber.model";


export class Planner {
  public id: string
  public wardName: string
  public presiding: Member
  public conducting: Member
  public openingHymn: Hymn
  public invocation: Member
  public wardBusiness: boolean //string
  public sacramentHymn: Hymn
  public sacramentPassing: boolean
  public talks: Talk[]
  public musicalNumber: MusicalNumber
  public closingHymn: Hymn
  public benediction: Member
  public dismissalSong: Hymn
  public sacramentDate: Date

  constructor(
    wardName: string = "",
    presiding: Member = null,
    conducting: Member = null,
    openingHymn: Hymn = null,
    invocation: Member = null,
    wardBusiness: boolean = false, // string,
    sacramentHymn: Hymn = null,
    sacramentPassing: boolean = false,
    talks: Talk[] = [],
    musicalNumber: MusicalNumber = null,
    closingHymn: Hymn = null,
    benediction: Member = null,
    dismissalSong: Hymn = null,
    sacramentDate: Date = null
  ) {

    this.wardName = wardName
    this.presiding = presiding ? presiding : new Member()
    this.conducting = conducting ? conducting : new Member()
    this.openingHymn = openingHymn ? openingHymn : new Hymn()
    this.invocation = invocation ? invocation : new Member()
    this.wardBusiness = wardBusiness
    this.sacramentHymn = sacramentHymn ? sacramentHymn : new Hymn()
    this.sacramentPassing = sacramentPassing
    this.talks = talks
    this.musicalNumber = musicalNumber ? musicalNumber : new MusicalNumber()
    this.closingHymn = closingHymn ? closingHymn : new Hymn()
    this.benediction = benediction ? benediction : new Member()
    this.dismissalSong = dismissalSong ? dismissalSong : new Hymn()
    this.sacramentDate = sacramentDate ? sacramentDate : new Date()
  }

  // get sacramentDate(): Date {
  //   return this.sacramentDate
  // }

  // set sacramentDate(sacramentDate: Date) {
  //   this.sacramentDate = sacramentDate
  // }
}
